azurerm/internal/services/sentinel/sentinel_data_connector_microsoft_cloud_app_security.go

```diff
@@ -0,0 +1,229 @@
+			TenantID: &tenantId,
+			DataTypes: &securityinsight.MCASDataConnectorDataTypes{
+				Alerts: &securityinsight.AlertsDataTypeOfDataConnectorAlerts{
+					State: alertState,
+				},
+				DiscoveryLogs: &securityinsight.MCASDataConnectorDataTypesDiscoveryLogs{
+					State: discoveryLogsState,
+				},
+			},
+		},
+		Kind: securityinsight.KindMicrosoftCloudAppSecurity,
+	}
+
+	// Service avoid concurrent updates of this resource via checking the "etag" to guarantee it is the same value as last Read.
+	// TODO: following code can be removed once the issue below is fixed:
+	// https://github.com/Azure/azure-rest-api-specs/issues/13203
+	if !d.IsNewResource() {
+		resp, err := client.Get(ctx, id.ResourceGroup, OperationalInsightsResourceProvider, id.WorkspaceName, name)
+		if err != nil {
+			return fmt.Errorf("retrieving %s: %+v", id, err)
+		}
+
+		dc, ok := resp.Value.(securityinsight.MCASDataConnector)
+		if !ok {
+			return fmt.Errorf("%s was not a Microsoft Cloud App Security Data Connector", id)
+		}
+		param.Etag = dc.Etag
+	}
+
+	if _, err = client.CreateOrUpdate(ctx, id.ResourceGroup, OperationalInsightsResourceProvider, id.WorkspaceName, id.Name, param); err != nil {
+		return fmt.Errorf("creating %s: %+v", id, err)
+	}
+
+	d.SetId(id.ID())
+
+	return resourceSentinelDataConnectorMicrosoftCloudAppSecurityRead(d, meta)
+}
+
+func resourceSentinelDataConnectorMicrosoftCloudAppSecurityRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Sentinel.DataConnectorsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.DataConnectorID(d.Id())
+	if err != nil {
+		return err
+	}
+	workspaceId := loganalyticsParse.NewLogAnalyticsWorkspaceID(id.SubscriptionId, id.ResourceGroup, id.WorkspaceName)
+
+	resp, err := client.Get(ctx, id.ResourceGroup, OperationalInsightsResourceProvider, id.WorkspaceName, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] %s was not found - removing from state!", id)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+
+	dc, ok := resp.Value.(securityinsight.MCASDataConnector)
+	if !ok {
+		return fmt.Errorf("%s was not a Microsoft Cloud App Security Data Connector", id)
+	}
+
+	d.Set("name", id.Name)
+	d.Set("log_analytics_workspace_id", workspaceId.ID())
+	d.Set("tenant_id", dc.TenantID)
+	if dt := dc.DataTypes; dt != nil {
+		alertsEnabled := false
+		if alert := dt.Alerts; alert != nil {
+			alertsEnabled = strings.EqualFold(string(alert.State), string(securityinsight.Enabled))
+			d.Set("alerts_enabled", alertsEnabled)
+		}
+
+		discoveryLogsEnabled := false
+		if discoveryLogs := dt.DiscoveryLogs; discoveryLogs != nil {
+			discoveryLogsEnabled = strings.EqualFold(string(discoveryLogs.State), string(securityinsight.Enabled))
+			d.Set("discovery_logs_enabled", discoveryLogsEnabled)
+		}
```

Comment:
the `Set` statement here for both AlertsEnabled and DiscoveryLogsEnabled needs to be moved outside of the if statement here